models/repo_pulse.go

```diff
@@ -0,0 +1,223 @@
+func (stats *PulseStats) MergedPRPerc() int {
+	return int(float32(stats.MergedPRCount()) / float32(stats.ActivePRCount()) * 100.0)
+}
+
+// ActiveIssueCount returns total active issue count
+func (stats *PulseStats) ActiveIssueCount() int {
+	return stats.OpenedIssueCount() + stats.ClosedIssueCount()
+}
+
+// OpenedIssueCount returns open issue count
+func (stats *PulseStats) OpenedIssueCount() int {
+	return len(stats.OpenedIssues)
+}
+
+// OpenedIssuePerc returns open issue count percent from total active
+func (stats *PulseStats) OpenedIssuePerc() int {
+	return int(float32(stats.OpenedIssueCount()) / float32(stats.ActiveIssueCount()) * 100.0)
+}
+
+// ClosedIssueCount returns closed issue count
+func (stats *PulseStats) ClosedIssueCount() int {
+	return len(stats.ClosedIssues)
+}
+
+// ClosedIssuePerc returns closed issue count percent from total active
+func (stats *PulseStats) ClosedIssuePerc() int {
+	return int(float32(stats.ClosedIssueCount()) / float32(stats.ActiveIssueCount()) * 100.0)
+}
+
+// UnresolvedIssueCount returns unresolved issue and pull request count
+func (stats *PulseStats) UnresolvedIssueCount() int {
+	return len(stats.UnresolvedIssues)
+}
+
+// FillPullRequestsForPulse returns pull request information for pulse page
+func FillPullRequestsForPulse(stats *PulseStats, baseRepoID int64, fromTime time.Time) error {
+	var err error
+	row := &struct {
+		Count int64
+	}{}
+
+	// Merged pull requests
+	sess := pullRequestsForPulseStatement(baseRepoID, fromTime, true, false)
+	sess.OrderBy("pull_request.merged_unix DESC")
+	stats.MergedPRs = make(PullRequestList, 0)
+	if err = sess.Find(&stats.MergedPRs); err != nil {
+		return err
+	}
+	if err = stats.MergedPRs.LoadAttributes(); err != nil {
+		return err
+	}
+
+	// Merged pull request authors
+	sess = pullRequestsForPulseStatement(baseRepoID, fromTime, true, false)
+	if _, err = sess.Select("count(distinct issue.poster_id) as `count`").Table("pull_request").Get(row); err != nil {
+		return err
+	}
+	stats.MergedPRAuthorCount = row.Count
+
+	// Opened pull requests
+	sess = pullRequestsForPulseStatement(baseRepoID, fromTime, false, true)
+	sess.OrderBy("issue.created_unix ASC")
+	stats.OpenedPRs = make(PullRequestList, 0)
+	if err := sess.Find(&stats.OpenedPRs); err != nil {
+		return err
+	}
+	if err = stats.OpenedPRs.LoadAttributes(); err != nil {
+		return err
+	}
+
+	// Opened pull request authors
+	row.Count = 0
+	sess = pullRequestsForPulseStatement(baseRepoID, fromTime, false, true)
+	if _, err = sess.Select("count(distinct issue.poster_id) as `count`").Table("pull_request").Get(row); err != nil {
+		return err
+	}
+	stats.OpenedPRAuthorCount = row.Count
+
+	return nil
+}
+
+func pullRequestsForPulseStatement(baseRepoID int64, fromTime time.Time, merged, created bool) *xorm.Session {
+	sess := x.Where("pull_request.base_repo_id=?", baseRepoID).
+		Join("INNER", "issue", "pull_request.issue_id = issue.id")
+
+	var mergedCond, createdCond builder.Cond
```

Comment:
I believe this can be simplified
```
var cond builder.Cond
if merged {
	cond = cond.Or(...)
}
if created {
	cond = cond.Or(...)
}
sess.And(cond)
```